sdk/python/lib/pulumi/runtime/settings.py

```diff
@@ -98,6 +95,73 @@ def __init__(
         else:
             self.engine = None
 
+    def __repr__(self):
+        return f"Settings: {self.engine}:{self.monitor}:{self.project}:{self.stack}"
+
+    @property
+    def engine(self):
+        return self._engine.get()
+
+    @engine.setter
+    def engine(self, v: Optional[Union[engine_pb2_grpc.EngineStub, Any]]):
+        return self._engine.set(v).var.get()
```

Comment:
`.set()` returns a `Token` which references the original `ContextVar` via `.var`, so we have to then `.get()` the new value :)